FGEGraphics/GraphicsHelpers/Models/ShapeGenerators.cs

```diff
@@ -0,0 +1,254 @@
+                float cosPhi = (float)Math.Cos(phi);
+                float x = radius * cosPhi * sinTheta;
+                float y = radius * cosTheta;
+                float z = radius * sinPhi * sinTheta;
+                vertices.Add(new Vector3(x, y, z));
+                normals.Add(new Vector3(x, y, z).Normalized());
+                texCoords.Add(new Vector2((float)j / slices, (float)i / stacks));
+                if (i < stacks && j < slices)
+                {
+                    uint currentRow = i * (slices + 1);
+                    uint nextRow = (i + 1) * (slices + 1);
+                    indices[index++] = currentRow + j;
+                    indices[index++] = nextRow + j;
+                    indices[index++] = currentRow + j + 1;
+                    indices[index++] = currentRow + j + 1;
+                    indices[index++] = nextRow + j;
+                    indices[index++] = nextRow + j + 1;
+                }
+            }
+        }
+        return GetModelAfterGenerating(modelEngine, "sphere", vertices, normals, texCoords, indices);
+    }
+
+    /// <summary>Generates a 2D circle model.</summary>
+    public static Model Generated2DCircle(float radius, uint slices, ModelEngine modelEngine, bool flip = false)
+    {
+        uint vertexCount = slices;
+        uint numIndices = slices * 3;
+        List<Vector3> vertices = new((int)vertexCount);
+        List<Vector3> normals = new((int)vertexCount);
+        List<Vector2> texCoords = new((int)vertexCount);
+        uint[] indices = new uint[numIndices];
+        int index = 0;
+        for (uint i = 0; i <= slices; i++)
+        {
+            float phi = i * 2 * MathHelper.Pi / slices;
+            float sinPhi = (float)Math.Sin(phi);
+            float cosPhi = (float)Math.Cos(phi);
+            float x = radius * cosPhi;
+            float y = radius * sinPhi;
+            float z = 0;
+            vertices.Add(new Vector3(x, y, z));
+            normals.Add(new Vector3(0, 0, flip ? -1 : 1));
+            texCoords.Add(new Vector2(cosPhi, sinPhi));
+            if (i < slices)
+            {
+                indices[index++] = 0;
+                indices[index++] = flip ? i + 1 : i + 2;
+                indices[index++] = flip ? i + 2 : i + 1;
+            }
+        }
+        return GetModelAfterGenerating(modelEngine, "circle", vertices, normals, texCoords, indices);
+    }
+
+    /// <summary>Generates a 3D cylinder model.</summary>
+    public static Model GenerateCylinder(float radius, float height, uint slices, uint stacks, ModelEngine modelEngine)
+    {
+        uint vertexCount = (stacks + 1) * (slices + 1);
+        uint numIndices = (stacks * slices * 6) * 2;
+        List<Vector3> vertices = new((int)vertexCount);
+        List<Vector3> normals = new((int)vertexCount);
+        List<Vector2> texCoords = new((int)vertexCount);
+        uint[] indices = new uint[numIndices];
+        int index = 0;
+        for (uint i = 0; i <= slices; i++)
```

Comment:
Given there are 3 separate circle generations in this class, should probably abstract em all to one shared internal method?

Reply:
Implemented in https://github.com/FreneticLLC/FreneticGameEngine/commit/eb11317057c23bb91a2bcdc2dbcc76db965c79f7